crates/turbopack-ecmascript/src/analyzer/mod.rs

```diff
@@ -1336,9 +1491,59 @@ impl JsValue {
                 LogicalOperator::And => all_if_known(list, JsValue::is_truthy),
                 LogicalOperator::Or => any_if_known(list, JsValue::is_truthy),
                 LogicalOperator::NullishCoalescing => {
-                    shortcircut_if_known(list, JsValue::is_not_nullish, JsValue::is_truthy)
+                    shortcircuit_if_known(list, JsValue::is_not_nullish, JsValue::is_truthy)
                 }
             },
+            JsValue::Binary(_, box a, op, box b) => {
+                let negate = matches!(
+                    op,
+                    BinaryOperator::NotEqual | BinaryOperator::StrictNotEqual
+                );
+                let positive_op = match op {
+                    BinaryOperator::NotEqual => BinaryOperator::Equal,
+                    BinaryOperator::StrictNotEqual => BinaryOperator::StrictEqual,
+                    _ => *op,
+                };
```

Comment:
This could be a struct method, which returns `(bool, PositiveBinaryOperator)`. That way we don't keep `BinaryOperator::*Not*` in the mix in the matches below, which we know aren't possible at this point.